services/httpoverrpc/httpoverrpc.proto

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+syntax = "proto3";
+
+option go_package = "github.com/Snowflake-Labs/sansshell/httpoverrpc";
+
+package HTTPOverRPC;
+
+// The HTTPOverRPC service definition
+service HTTPOverRPC {
+  // Make an HTTP call to localhost
+  rpc Localhost(LocalhostHTTPRequest) returns (HTTPReply) {}
+}
+
+message LocalhostHTTPRequest {
+  HTTPRequest request = 1;
+  // The port to use for the request on the local host.
+  int32 port = 2;
+}
+
+message Header {
+  string key = 1;
+  repeated string values = 2;
+}
+
+// HTTPRequest describes the HTTP request
+message HTTPRequest {
+  string method = 1;
```

Comment:
for this and 'headers', I wonder about canonicalization and the intersection with policy enforcement. A good policy should naturally not say things like "allow anything where method is not POST", but I could see some confusion since in the sanssh client, "--method" is a flag, so you could pass either --method POST or --method post or --method Post. All of them would "work", but this might make the policy difficult to write.

Reply:
The two options I like:

1. Accept that this can be finicky to use and err towards passing the user's input directly into the RPC call
2. Canonicalize the user's inputs inside the client before passing them into the RPC

Which do you think would be better? I'm leaning a bit towards the implementation simplicity of 1.

Reply:
1. is _probably_ fine, if we think we can avoid the policy footguns. 6 months from now, I'm not sure I'd instinctively recognize a policy like:
```
allow { input.method not in ['post', 'delete'] } 
```
as potentially dangerous.

2 probably isn't even the right fix if we're really worried about this. We'd have to canonicalize in the proxy before policy eval, right? (Which we could theoretically do with a authz hook)

Reply:
I've typically seen sansshell policies use allowlists when the set of possible values is small and only switch to blocklists when it's hard to enumerate allowed values. What you mention is a footgun but it doesn't seem much worse than other commands that allow string inputs.

I'd rather not add a precedent of the proxy tampering with input values passed in by the user.

One more option is to switch method to an enum. I think the flexibility of a string is better, but I could be convinced otherwise.

Reply:
I agree that switching to an enum isn't a good option, and wouldn't work for headers, unless we really wanted to limit the functionality.

I don't know that I'd agree that canonicalizing the input before policy eval is "tampering", since that's what the authz hooks are for, in a sense, but I'm ok deferring that discussion for now